gulpfile.js

```diff
@@ -6,12 +6,28 @@ const sass = require('gulp-sass');
 const svgSprite = require('gulp-svg-sprite');
 const swBuild = require('sw-build');
 const fs = require('fs');
+const path = require('path');
+const abe = require('amp-by-example');
+
+const config = {
+  src: path.join(__dirname, 'examples/src'), // root folder containing the examples
+  destRoot: path.join(__dirname, 'build'), // target dir for generated embeds
+  destDir: '/examples', // optional sub dir
+  host: 'https://ampproject-b5f4c.firebaseapp.com' // from where the embeds are going to be served
+}
 
 const Path = {
   CSS_SOURCES: './assets/sass/**/*.scss',
   CSS_OUT_DIR: './assets/css/'
 };
 
+// build examples used as sample embeds in docs
+gulp.task('build-examples', function() {
+  abe.generatePreview(config);
```

Comment:
The call is synchronous, but runs other async calls. I need to properly promisify the API.